sympy/parsing/latex/lark/latex.lark

```diff
@@ -133,36 +135,67 @@ group: L_PAREN expression R_PAREN
     | L_BRACE expression R_BRACE
     | L_BRACE_LITERAL expression R_BRACE_LITERAL
 
-abs_group: BAR expression BAR
-floor_group: L_FLOOR expression R_FLOOR
-ceil_group: L_CEIL expression R_CEIL
+group_round_parentheses: L_PAREN expression R_PAREN
+group_curly_parentheses: L_BRACE expression R_BRACE
 
-relation: atom (EQUAL | NOT_EQUAL | LT | LTE | GT | GTE) atom
+relation: expression_core (EQUAL | NOT_EQUAL | LT | LTE | GT | GTE) expression_core
+
+expression_core: atom
+  | group_round_parentheses
+  | group_curly_parentheses
 
 add: expression ADD expression
 sub: expression SUB expression
-mul: expression MUL expression | (oneletter_symbol oneletter_symbol | number oneletter_symbol)
-div: expression DIV expression
+mul: expression MUL_SYMBOL expression
+    | (oneletter_symbol oneletter_symbol | number oneletter_symbol)
+div: expression DIV_SYMBOL expression
 
 group_paren: L_PAREN expression R_PAREN
 
-expression: atom
+expression: expression_core
     | add
     | sub
     | mul
     | div
-    | expression_curly
     | fraction
+    | binomial
+    | function
     | ambiguous_sub
-    | group_paren
 
 ambiguous_sub: oneletter_symbol number
 
-expression_curly: L_BRACE expression R_BRACE
+fraction: basic_fraction
+    | simple_fraction
+    | general_fraction
+
+general_fraction.0: CMD_FRAC group_curly_parentheses group_curly_parentheses expression?
+
+simple_fraction.1: CMD_FRAC DIGIT group_curly_parentheses expression?
+    | CMD_FRAC group_curly_parentheses DIGIT expression?
+
+basic_fraction.2: CMD_FRAC DIGIT DIGIT expression?
+
+function: abs | floor | ceil
+    | trigonometric_function
+
+trigonometric_function: sin //| cos | tan | csc | sec | tan
+
+abs: BAR expression BAR
+floor: L_FLOOR expression R_FLOOR
+ceil: L_CEIL expression R_CEIL
+
+sin: FUNC_SIN expression
 
-fraction: CMD_FRAC expression expression
+binomial: basic_binomial
+    | simple_binomial
+    | general_binomial
 
-binomial: CMD_BINOM L_BRACE group R_BRACE L_BRACE group R_BRACE
+general_binomial.0: CMD_BINOM group_curly_parentheses group_curly_parentheses expression?
```

Comment:
We should create two kinds of expressions: the ones that have higher priority than `\binom` and the ones that have lower. That was the idea of `expression_core`... `expression_core` should contain all expressions that have higher priority than the backslash operators.

Generic `expression` should be either an `expression_core` or any kind of expressions that have lower priority than backslash operators.

This way should work.

Reply:
Using a named rule `A` in a different rule `B` does not mean that `A` has a higher priority than `B`. That is to say, `expression_core` does not have a higher precedence than `expression`. Both have equal precedence, which is the root reason for many of the ambiguous parses.

Assigning `expression_core` a higher precedence than `expression` _is_ possible, but there are two things to keep in mind:
1. Assigning priorities to rules is only possible using the Earley parser and not the LALR parser. Since you mentioned that you think that moving to the LALR parser might be a better option, I am mentioning this.
2. If we go with the rule priorities route, then we lose the ability to inform the user that their input expression is ambiguous. Using rule priorities requires to use `ambiguity=resolve`, which means that any ambiguity in the parsing is resolved automatically (using rule priorities, if that is available). If we keep `ambiguity=explicit`, we retain the ability to inform the user of ambiguous parses, but lose the ability to use rules with different priorities.

Reply:
I'm also marking this as resolved, because we discussed how to solve this problem during the team meeting, and the way to solve this is to use the same strategy but for binomial expressions.